hack/jenkins/e2e-jobs.sh

```diff
@@ -14,9 +14,1105 @@
+    KUBEMARK_MASTER_SIZE="n1-standard-16"
+    KUBEMARK_NUM_NODES="1000"
+    ;;
+esac
+
+# AWS variables
+export KUBE_AWS_INSTANCE_PREFIX=${E2E_CLUSTER_NAME}
+export KUBE_AWS_ZONE=${E2E_ZONE}
+export AWS_CONFIG_FILE=${AWS_CONFIG_FILE:-}
+export AWS_SSH_KEY=${AWS_SSH_KEY:-}
+export KUBE_SSH_USER=${KUBE_SSH_USER:-}
+export AWS_SHARED_CREDENTIALS_FILE=${AWS_SHARED_CREDENTIALS_FILE:-}
+
+# GCE variables
+export INSTANCE_PREFIX=${E2E_CLUSTER_NAME}
+export KUBE_GCE_ZONE=${E2E_ZONE}
+export KUBE_GCE_NETWORK=${E2E_NETWORK}
+export KUBE_GCE_INSTANCE_PREFIX=${KUBE_GCE_INSTANCE_PREFIX:-}
+export KUBE_GCS_STAGING_PATH_SUFFIX=${KUBE_GCS_STAGING_PATH_SUFFIX:-}
+export KUBE_GCE_NODE_PROJECT=${KUBE_GCE_NODE_PROJECT:-}
+export KUBE_GCE_NODE_IMAGE=${KUBE_GCE_NODE_IMAGE:-}
+export KUBE_OS_DISTRIBUTION=${KUBE_OS_DISTRIBUTION:-}
+export GCE_SERVICE_ACCOUNT=$(gcloud auth list 2> /dev/null | grep active | cut -f3 -d' ')
+
+# GKE variables
+export CLUSTER_NAME=${E2E_CLUSTER_NAME}
+export ZONE=${E2E_ZONE}
+export KUBE_GKE_NETWORK=${E2E_NETWORK}
+export E2E_SET_CLUSTER_API_VERSION=${E2E_SET_CLUSTER_API_VERSION:-}
+export CMD_GROUP=${CMD_GROUP:-}
+export MACHINE_TYPE=${NODE_SIZE:-}  # GKE scripts use MACHINE_TYPE for the node vm size
+
+if [[ ! -z "${GKE_API_ENDPOINT:-}" ]]; then
+  export CLOUDSDK_API_ENDPOINT_OVERRIDES_CONTAINER=${GKE_API_ENDPOINT}
+fi
+
+# Shared cluster variables
+export E2E_MIN_STARTUP_PODS=${E2E_MIN_STARTUP_PODS:-}
+export KUBE_ENABLE_CLUSTER_MONITORING=${ENABLE_CLUSTER_MONITORING:-}
+export KUBE_ENABLE_CLUSTER_REGISTRY=${ENABLE_CLUSTER_REGISTRY:-}
+export KUBE_ENABLE_HORIZONTAL_POD_AUTOSCALER=${ENABLE_HORIZONTAL_POD_AUTOSCALER:-}
+export KUBE_ENABLE_DEPLOYMENTS=${ENABLE_DEPLOYMENTS:-}
+export KUBE_ENABLE_EXPERIMENTAL_API=${ENABLE_EXPERIMENTAL_API:-}
+export MASTER_SIZE=${MASTER_SIZE:-}
+export NODE_SIZE=${NODE_SIZE:-}
+export NODE_DISK_SIZE=${NODE_DISK_SIZE:-}
+export NUM_NODES=${NUM_NODES:-}
+export TEST_CLUSTER_LOG_LEVEL=${TEST_CLUSTER_LOG_LEVEL:-}
+export TEST_CLUSTER_RESYNC_PERIOD=${TEST_CLUSTER_RESYNC_PERIOD:-}
+export PROJECT=${PROJECT:-}
+export NETWORK_PROVIDER=${NETWORK_PROVIDER:-}
+export JENKINS_PUBLISHED_VERSION=${JENKINS_PUBLISHED_VERSION:-'ci/latest'}
+
+export KUBE_ADMISSION_CONTROL=${ADMISSION_CONTROL:-}
+
+export KUBERNETES_PROVIDER=${KUBERNETES_PROVIDER}
+export PATH=${PATH}:/usr/local/go/bin
+export KUBE_SKIP_UPDATE=y
+export KUBE_SKIP_CONFIRMATIONS=y
 
-echo "e2e-jobs.sh not implemented yet. See #18981 for details."
+# E2E Control Variables
+export E2E_OPT="${E2E_OPT:-}"
+export E2E_UP="${E2E_UP:-true}"
+export E2E_TEST="${E2E_TEST:-true}"
+export E2E_DOWN="${E2E_DOWN:-true}"
+export E2E_CLEAN_START="${E2E_CLEAN_START:-}"
+# Used by hack/ginkgo-e2e.sh to enable ginkgo's parallel test runner.
+export GINKGO_PARALLEL=${GINKGO_PARALLEL:-}
+export GINKGO_TEST_ARGS="${GINKGO_TEST_ARGS:-}"
 
+bash <(curl -fsS --retry 3 "https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/jenkins/e2e-runner.sh")
```

Comment:
If we make it one of the files it should probably be the config, since all the current jobs call it. In fact, that might be ideal. We'd get to keep the history and we won't have to touch any job configs.

I can make it source the file instead of bash. Eventually we should use bash and turn off xtrace so that our console output isn't so noisy.